news-explorer-frontend/src/utils/MainApi.js

```diff
@@ -0,0 +1,114 @@
+export const BASE_URL = "https://www.namernews.students.nomoredomainssbs.ru/"; //"http://localhost:3001"   //
+
+class MainApi {
+
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+
+  _checkResponse(res){
+    return res.ok ? res.json() : Promise.reject(`Error: ${res.StatusText}`)
+  }
+
+// async _connect(url, headers) {
+//   const res = await fetch(url, headers);
+//     return res.ok ? res.json() : await Promise.reject(`Error: ${res.status}`)
+//  }
+
+  signup({name, email, password}) {
+    console.log(name, email, password)
+    return fetch(`${BASE_URL}/signup`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify({name, email, password}),
+    })
+    .then(this._checkResponse)

```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.

Reply:
Updated w.r.t your suggestion +